src/plugins/observables.js

```diff
@@ -1,9 +1,40 @@
 /* @flow */
 import isObservable from 'is-observable'
 import StyleRule from '../rules/StyleRule'
-import type {Rule} from '../types'
+import createRule from '../utils/createRule'
+import type {Observable, Rule, RuleOptions, JssStyle} from '../types'
 
 export default {
+  onCreateRule(name: string, decl: JssStyle, options: RuleOptions): Rule | null {
+    if (isObservable(decl)) {
+      // Cast `decl` to `Observable`, since it passed the type guard
+      const props$ = (decl: Observable<{[string]: string}>);
+
+      // We know `rule` is a `StyleRule`, and the other types don't have a
+      // `prop` method, so we must explicitly cast to `StyleRule`
+      const rule = ((createRule(name, {}, options): any): StyleRule);
```

Comment:
No idea how this works, you managed to cast to any and then to StyleRule haha

Reply:
Common pattern when you know more about the state of the system than the type-checker: cast to `any`, then cast to the correct type.  It's dangerous if you get it wrong though (you're explicitly telling Flow to ignore what looks like a type error) - it's effectively `!important` for types.

I believe Microsoft actually suggests this pattern in [the TypeScript docs](https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions).

Reply:
learned something new! thanks